File: crates/zyntax_python/pressure/closures.py
# A lambda over a loop variable made and called every step.
import sys

def apply(f, v: int) -> int:
    return f(v)

def main(n: int) -> int:
    total = 0
    for i in range(n):
        f = lambda x: x + i
        total += apply(f, 1)
    return total

print(main(int(sys.argv[1])))

File: crates/zyntax_lua/pressure/closures.lua
-- A closure over the loop variable made and called every step.
local function apply(f, v)
  return f(v)
end

local function main(n)
  local total = 0
  for i = 1, n do
    local f = function(x) return x + i end
    total = total + apply(f, 1)
  end
  return total
end

print(main(tonumber(arg[1])))
